BugsAndExploits/TntDuping.md

```diff
@@ -6,15 +7,21 @@ description: Explaination of tnt duplication
+It gained popularity with the invention of world eaters.
+### Opposition
+Mojang historically opposes TNT Duping as they consider it an exploit, and tried to fix it, but then have decided to leave it in the game for the time being, due to a lack of replacement method
+https://www.reddit.com/r/Minecraft/comments/fkt9jf/anchor_yourself_to_the_nether_snapshot_20w12a_is/fkv33q3?utm_source=share&utm_medium=web2x&context=3
+ for what TNT duping currently does.
 ## Mechanic
 TNT Duping works by using piston pushing behavior to make the TNT both light and move at once, creating a moving TNT block and a TNT entity. The order this follows is:
 The piston, in the same tick it powers but before moving, creates a list of the blocks in front of it that need to be moved.
-The TNT gets powered. In order to do this, the update that powers it needs to come from a moving rail (1.12 and below), a moving coral fan (1.13+)(specifically a moving block replacing the block supporting the coral fan), or a moving lit observer (works in all versions that have observers), which needs to happen before the TNT block is removed (before a block behind the tnt replaces it, or if there is no blocks, it can be after the moving TNT was created). 
-These specific blocks are able to update after the list of blocks to be moved is created, but before the blocks have moved. This update can also be propagated through other blocks, as long as it is instant (for example using rails, dust, noteblocks (1.13+), or TNT itself), which allows for up to 11 TNT to be duplicated using a single piston push, the 12th block being for updating.
-In this step, the TNT entity has already been created from step 2, so now the piston actually moves the blocks it already has in the list, and this means that the TNT is preserved in block form.
-Opposition
-Mojang historically opposes TNT Duping as they consider it an exploit, but have decided to leave it in the game for the time being, 
-due to a lack of replacement method
-https://www.reddit.com/r/Minecraft/comments/fkt9jf/anchor_yourself_to_the_nether_snapshot_20w12a_is/fkv33q3?utm_source=share&utm_medium=web2x&context=3
- for what TNT duping currently does.
\ No newline at end of file
+The TNT gets powered. In order to do this, the update that powers it needs to come from
+- a moving rail (1.12 and below) being replaced
+- anything that pop off from a state update (supporting block removed): coral fan is mostly used because it still create a moving block and can be reused.
+If it's a brittle block, that block must not be in front of a moved block because it will be placed in the broken block list that will be handled separately
+- a moving lit observer being replaced (works in all versions that have observers)
+
+These specific blocks are able to create a normal update after the list of blocks to be moved is created, but before some of the moved blocks have been removed.
```

Comment:
they work cus they give an update when moved or when something it is next too moves

Reply:
it's the old blocks that cause updates, I precise here if it's a brittle block poping off (caused by a nearby block replacing the support block), or special updates when the state is replaced, I should expand the observer case to most components that can hard power (except detector rail)

Reply:
You could just write the order in which each action is done for pistons. Then explain why duping happens at the specific point

Reply:
the actions are already detailed in the piston page,I should also put a link
redstone torch is included in components that hard power